src/main/java/org/folio/circulation/infrastructure/storage/inventory/ItemRepository.java

```diff
@@ -254,6 +257,20 @@ private CompletableFuture<Result<MultipleRecords<Item>>> fetchItems(Collection<S
     final var mapper = new ItemMapper();
 
     return finder.findByIds(itemIds)
+      .thenApply(mapResult(identityMap::add))
+      .thenApply(mapResult(records -> records.mapRecords(mapper::toDomain)))
+      .thenCompose(res -> res.value().getTotalRecords() == itemIds.size()
+        ? CompletableFuture.completedFuture(res)
+        : res.combineAfter(x -> lookupDcbItem(res, itemIds), MultipleRecords::combine));
+  }
+
+  private CompletableFuture<Result<MultipleRecords<Item>>> lookupDcbItem(Result<MultipleRecords<Item>> inventoryItems, Collection<String> itemIds) {
```

Comment:
This line seems to exceed 100 characters limit. Also, can you please add some logging , at least for method start? https://wiki.folio.org/display/DD/Logging